nexus/comment.ts

```diff
@@ -0,0 +1,384 @@
+            id: args.postId,
+          },
+          include: {
+            author: true,
+          },
+        })
+
+        if (!post) {
+          throw new NotFoundError('post')
+        }
+
+        const postComment = await ctx.db.postComment.create({
+          data: {
+            body: args.body,
+            author: {
+              connect: { id: userId },
+            },
+            post: {
+              connect: { id: post.id },
+            },
+          },
+          include: {
+            author: true,
+          },
+        })
+
+        if (postComment.author.id !== post.author.id) {
+          await transport.sendMail({
+            from: 'robin@journaly.com',
+            to: post.author.email,
+            subject: "You've got feedback!",
+            html: makeEmail(`
+              <p>Great news! <strong>@${postComment.author.handle}</strong> left you some feedback!</p>
+              <p><strong>Journal entry:</strong> ${post.title}</p>
+              <p><strong>Comment:</strong> "${postComment.body}"</p>
+              <p>Click <a href="https://${process.env.SITE_DOMAIN}/post/${post.id}">here</a> to go to your journal entry!</p>
+            `),
+          })
+        }
+
+        // TODO: Set up logging and check for successful `mailResponse`
+        return postComment
+      },
+    })
+
+    t.field('updatePostComment', {
+      type: 'PostComment',
+      args: {
+        postCommentId: intArg({ required: true }),
+        body: stringArg({ required: true }),
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) throw new Error('You must be logged in to do that.')
+
+        const [currentUser, originalPostComment] = await Promise.all([
+          ctx.db.user.findOne({
+            where: {
+              id: userId,
+            },
+          }),
+          ctx.db.postComment.findOne({
+            where: {
+              id: args.postCommentId,
+            },
+          }),
+        ])
+
+        if (!currentUser) throw new Error('User not found.')
+        if (!originalPostComment) throw new Error('Comment not found.')
+
+        hasPostPermissions(originalPostComment, currentUser)
+
+        const postComment = await ctx.db.postComment.update({
+          data: {
+            body: args.body,
+          },
+          where: {
+            id: args.postCommentId,
+          },
+        })
+
+        return postComment
+      },
+    })
+    t.field('deletePostComment', {
+      type: 'PostComment',
+      args: {
+        postCommentId: intArg({ required: true }),
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) throw new Error('You must be logged in to do that.')
+
+        const currentUser = await ctx.db.user.findOne({
+          where: {
+            id: userId,
+          },
+        })
+
+        if (!currentUser) throw new Error('User not found.')
```

Comment:
We do this block of logic in a bunch of places. It would be nice to have a util function like `getLoggedInUser(ctx)` that returns the current user or throws if they're not logged in or can't be found in the DB

Reply:
Scheduled for a follow-up! ✅